onmt/utils/loss.py

```diff
@@ -261,6 +261,51 @@ def _bottle(self, _v):
     def _unbottle(self, _v, batch_size):
         return _v.view(-1, batch_size, _v.size(1))
 
+    def ignore_prompt(
+        self, batch, response_token="▁Response", response_right_pattern_length=3
+    ):
+        """
+        Mask the prompt in the target side of the bath examples in order
+            to set the loss of the prompt to zero.
+        For finetuning on specific tasks.
+        The masks are supposed to be properly handled by the loss criterion
+            (e.g. nn.CrossEntropyLoss )
```

Comment:
Do we want to add some kind of reference around here? (Ghost attention, other ?)

Reply:
Ghost attention is something else.
It is mentioned is various papers, like
- The Self-instruct paper (alpaca finetuning)
![image](https://github.com/l-k-11235/OpenNMT-py/assets/57141057/e7b78d6d-d555-46c0-9447-f72ca2afd9ea)
- The Llama-2 paper:
![image](https://github.com/l-k-11235/OpenNMT-py/assets/57141057/a12fb0a5-d16d-4af8-a7d2-de52a0f83ff0)

Reply:
The Llama2 paper is confusing. This 0 loss thing seems to be part of the Ghost Attention method, but indeed there might be other aspects to this.
![Screenshot from 2023-09-04 16-31-03](https://github.com/l-k-11235/OpenNMT-py/assets/14880403/f2a1c490-ce71-46a7-9e5f-2938ba0bd57c)